File: ccif/src/java/other/org/himalay/ccif/FirmwareVerReportM_Mode.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
// 
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.ccif ;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.lang.Exception;
import java.io.ByteArrayInputStream;
import java.io.PrintStream;
import org.himalay.msgs.runtime.Created;

import org.himalay.msgs.runtime.*;
@Created(date = "Fri Jan 15 01:33:05 EST 2016")

public  class FirmwareVerReportM_Mode extends BinStruct implements PublicBinMsg{




    // members variables
    // firmWareVersionCharacter
    public ByteArray firmWareVersionCharacter ;
    // fwareDate
    public FirmWareDate fwareDate ;

        public FirmwareVerReportM_Mode () // throws Exception
    {
            init();
    }
    
    private void init()
    {
        // Initialize firmWareVersionCharacter
        firmWareVersionCharacter= new ByteArray();firmWareVersionCharacter.setSizeType("FIRST_UI8");
        // Initialize fwareDate
        fwareDate = new FirmWareDate();
    }

    public int readNoHeader(DataInputStream istream) throws IOException 
    {
    
     return read(istream);
       }
   
    public int read(DataInputStream istream) throws IOException 
    {
    preRead();
        int retVal= 0;

                     // read firmWareVersionCharacter
        {retVal+=firmWareVersionCharacter.read(istream); }
        // read fwareDate
        retVal += fwareDate.read(istream);

        postRead();
        return retVal;
    }


    public int write(DataOutputStream ostream) throws IOException 
    {
    preWrite();
        int retVal= 0;

        { /** fix dependent sizes for firmWareVersionCharacter**/  }
        { /** fix dependent sizes for fwareDate **/  }
    
                // write firmWareVersionCharacter
        {retVal += firmWareVersionCharacter.write(ostream);}
        // write fwareDate
        if (fwareDate!=null)retVal +=fwareDate.write(ostream);
postWrite();
        return retVal;
    }
    
    public int dump(DumpContext dc) throws IOException 
    {
        dc.indent();dc.getPs().print("FirmwareVerReportM_Mode\n");
    dc.increaseIndent();
        int retVal= 0;
        // write firmWareVersionCharacter
        dc.indent();dc.getPs().print("firmWareVersionCharacter: "+firmWareVersionCharacter.getSize()+"(0x"+Integer.toHexString(firmWareVersionCharacter.getSize())+")\n");this.firmWareVersionCharacter.dump(dc);
        // write fwareDate
        if ( fwareDate != null ) {dc.indent();dc.getPs().println("fwareDate") ;retVal +=fwareDate.dump(dc);}
dc.decreaseIndent();
        return retVal;
    }


        // Getter for firmWareVersionCharacter
    //public ByteArray getFirmWareVersionCharacter()
    //{
    //    return firmWareVersionCharacter ;
    //}

    
    // Setter for firmWareVersionCharacter
    //public void setFirmWareVersionCharacter(ByteArray val)
    //{
    //    this.firmWareVersionCharacter= val;
    //}
    // Getter for fwareDate
    //public FirmWareDate getFwareDate()
    //{
    //    return fwareDate ;
    //}

    
    // Setter for fwareDate
    //public void setFwareDate(FirmWareDate val)
    //{
    //    this.fwareDate= val;
    //}


    public void setFirmWareVersionCharacter(byte[] val)
    {
        this.firmWareVersionCharacter.setData(val);
    }
    
    
    public int getSize() throws IOException
    {
       DataOutputStream dos= new DataOutputStream(new NullStream());
       return this.write(dos);
    }
    
        

}

// End of code

File: libccif/src/bp/FirmwareVerReportM_Mode.cpp
//**//
//@Created(date = "Mon Feb 01 22:30:17 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
//**//
#include "stdio.h"
#include "FirmwareVerReportM_Mode.h"

#ifdef org_himalay_ccif_FirmwareVerReportM_Mode__USE_SMART_PTR
#define ArrayList(x) SmartPtrList<x>
#define ArrayList_iterator(x) SmartPtrList<x>::iterator
#include "SmartPtrList.h"
#else
#define ArrayList(x) BinMessagePtrList
#define ArrayList_iterator(x) BinMessagePtrList::iterator
#include "BinMessagePtrList.h"
#endif

// Namespace
namespace org {
namespace himalay {
namespace ccif {

using namespace ::org::himalay::msgs::runtime;

// Constructor
FirmwareVerReportM_Mode::FirmwareVerReportM_Mode() {
    references= 0;


    // firmWareVersionCharacter
    org_himalay_msgs_runtime_ByteArray_NEW(firmWareVersionCharacter);
    firmWareVersionCharacter->setSizeType("FIRST_UI8");
    // fwareDate
    org_himalay_ccif_FirmWareDate__NEW( fwareDate);
}
// Destructor
FirmwareVerReportM_Mode::~FirmwareVerReportM_Mode() {
    // firmWareVersionCharacter
    org_himalay_msgs_runtime_ByteArray_DELETE( firmWareVersionCharacter);
    // fwareDate
    org_himalay_ccif_FirmWareDate__DELETE( fwareDate);
}

int FirmwareVerReportM_Mode::readNoHeader(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;

    // firmWareVersionCharacter
    {
        retVal+=firmWareVersionCharacter->read(istream);
    }
    // fwareDate
    retVal += fwareDate->read(istream);
    return retVal;
}

int FirmwareVerReportM_Mode::read(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;
    // read firmWareVersionCharacter
    {
        retVal+=firmWareVersionCharacter->read(istream);
    }
    // read fwareDate
    retVal += fwareDate->read(istream);

    return retVal;
}

int FirmwareVerReportM_Mode::write(DataOutputStream& ostream) { // throws IOException
    {   /** fix dependent sizes for firmWareVersionCharacter**/
    }
    {   /** fix dependent sizes for fwareDate **/
    }

    int retVal= 0;
    // write firmWareVersionCharacter
    {
        retVal += firmWareVersionCharacter->write(ostream);
    }
    // write fwareDate
    if ( fwareDate != NULL ) {
        retVal +=fwareDate->write(ostream);
    }

    return retVal;
}

int FirmwareVerReportM_Mode::dump(DumpContext& dc) { // throws IOException
    dc.indent();
    dc.getPs().print("FirmwareVerReportM_Mode\n");
    dc.increaseIndent();
    int retVal= 0;
    // write firmWareVersionCharacter
    dc.indent();
    dc.getPs().print("firmWareVersionCharacter");
    firmWareVersionCharacter->dump(dc);
    // write fwareDate
    if ( fwareDate != NULL ) {
        dc.indent();
        dc.getPs().println("fwareDate") ;
        retVal +=fwareDate->dump(dc);
    }
    dc.decreaseIndent();
    return retVal;
}


// Getter for firmWareVersionCharacter
//org_himalay_msgs_runtime_ByteArray_PTR_TYPE FirmwareVerReportM_Mode::getFirmWareVersionCharacter()
//{
//return this->firmWareVersionCharacter;
//}

// Setter for firmWareVersionCharacter
//void FirmwareVerReportM_Mode::setFirmWareVersionCharacter(org_himalay_msgs_runtime_ByteArray_PTR_TYPE val)
//{
//this->firmWareVersionCharacter= val;
//}

// Getter for fwareDate
//org_himalay_ccif_FirmWareDate__PTR_TYPE FirmwareVerReportM_Mode::getFwareDate()
//{
//return this->fwareDate;
//}

// Setter for fwareDate
//void FirmwareVerReportM_Mode::setFwareDate(org_himalay_ccif_FirmWareDate__PTR_TYPE val)
//{
//this->fwareDate= val;
//}

int FirmwareVerReportM_Mode::getSize() { // throws IOException
    int retVal = 0;

    return retVal;
}

#ifdef org_himalay_ccif_FirmwareVerReportM_Mode__USE_SMART_PTR
void intrusive_ptr_add_ref(org::himalay::ccif::FirmwareVerReportM_Mode* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::FirmwareVerReportM_Mode* p) {
    p->decreaseRef();
};;
#endif


}
}
}

// End of code